File: backend/src/main/java/com/supplier/sprsystem/service/UserService.java
package com.supplier.sprsystem.service;

import com.supplier.sprsystem.dto.request.PasswordChangeRequest;
import com.supplier.sprsystem.dto.request.UserRoleUpdateRequest;
import com.supplier.sprsystem.dto.request.UserUpdateRequest;
import com.supplier.sprsystem.dto.response.UserResponse;

import java.util.List;

public interface UserService {
    List<UserResponse> getAllUsers();
    UserResponse getUserById(Long id);
    UserResponse getUserByUsername(String username);
    List<UserResponse> searchUsers(String keyword);
    UserResponse updateUser(Long id, UserUpdateRequest updateRequest);
    UserResponse updateUserRoles(Long id, UserRoleUpdateRequest roleUpdateRequest);
    UserResponse toggleUserStatus(Long id);
    void changePassword(Long userId, PasswordChangeRequest passwordChangeRequest);
    void deleteUser(Long id);
}

File: frontend/src/services/user.service.js
import api from './api';

export const userService = {
  getAllUsers: async () => {
    const response = await api.get('/users');
    return response.data;
  },

  getUserById: async (id) => {
    const response = await api.get(`/users/${id}`);
    return response.data;
  },

  searchUsers: async (keyword) => {
    const response = await api.get('/users/search', { params: { keyword } });
    return response.data;
  },

  updateUser: async (id, userData) => {
    const response = await api.put(`/users/${id}`, userData);
    return response.data;
  },

  updateUserRoles: async (id, roles) => {
    const response = await api.put(`/users/${id}/roles`, { roles });
    return response.data;
  },

  toggleUserStatus: async (id) => {
    const response = await api.patch(`/users/${id}/toggle-status`);
    return response.data;
  },

  changePassword: async (id, passwordData) => {
    const response = await api.post(`/users/${id}/change-password`, passwordData);
    return response.data;
  },

  deleteUser: async (id) => {
    const response = await api.delete(`/users/${id}`);
    return response.data;
  }
};
